From 20f8ba9cfae4a3c70c3131d4f14bf14ba4dc5ced Mon Sep 17 00:00:00 2001
From: HRUSHIKESH DOKALA <96101829+Hk669@users.noreply.github.com>
Date: Sat, 22 Jun 2024 00:22:23 +0530
Subject: [PATCH] fix: created in ChatCompletion for clients (#2988)


diff --git a/autogen/oai/anthropic.py b/autogen/oai/anthropic.py
index 9f9203ef..0bbbec39 100644
--- a/autogen/oai/anthropic.py
+++ b/autogen/oai/anthropic.py
@@ -181,7 +181,7 @@ class AnthropicClient:
         response_oai = ChatCompletion(
             id=response.id,
             model=anthropic_params["model"],
-            created=int(time.time() * 1000),
+            created=int(time.time()),
             object="chat.completion",
             choices=choices,
             usage=CompletionUsage(
diff --git a/autogen/oai/gemini.py b/autogen/oai/gemini.py
index 30d4c3fe..8babb872 100644
--- a/autogen/oai/gemini.py
+++ b/autogen/oai/gemini.py
@@ -253,7 +253,7 @@ class GeminiClient:
         response_oai = ChatCompletion(
             id=str(random.randint(0, 1000)),
             model=model_name,
-            created=int(time.time() * 1000),
+            created=int(time.time()),
             object="chat.completion",
             choices=choices,
             usage=CompletionUsage(
diff --git a/autogen/oai/mistral.py b/autogen/oai/mistral.py
index 83236937..8017e353 100644
--- a/autogen/oai/mistral.py
+++ b/autogen/oai/mistral.py
@@ -175,7 +175,7 @@ class MistralAIClient:
         response_oai = ChatCompletion(
             id=mistral_response.id,
             model=mistral_response.model,
-            created=int(time.time() * 1000),
+            created=int(time.time()),
             object="chat.completion",
             choices=choices,
             usage=CompletionUsage(
diff --git a/autogen/oai/together.py b/autogen/oai/together.py
index ef6c0980..bbbe851b 100644
--- a/autogen/oai/together.py
+++ b/autogen/oai/together.py
@@ -209,7 +209,7 @@ class TogetherClient:
         response_oai = ChatCompletion(
             id=response.id,
             model=together_params["model"],
-            created=int(time.time() * 1000),
+            created=int(time.time()),
             object="chat.completion",
             choices=choices,
             usage=CompletionUsage(
